app/src/main/java/org/xbmc/kore/service/library/SyncMovies.java

```diff
@@ -163,7 +163,6 @@ public void onSuccess(List<VideoType.DetailsMovie> result) {
                     // As we're going to recurse, these result objects can add up, so
                     // let's help the GC and indicate that we don't need this memory
                     // (hopefully this works)
-                    result = null;
```

Comment:
Did you read the comment above this statement?
Are you sure setting result to null is of no use at all?